src/macros.jl

```diff
@@ -0,0 +1,53 @@
+import Test: get_testset
+
+"""
+    @recordproperty name::AbstractString value::Union{AbstractString, Expr}
+
+Adds a property to a testset with `name` and `value` that will in turn be added
+to the `<properties>` node of the corresponding testsuite in the JUnit XML.
+`value` must either be a `String` or an `Expr` that evaluates to a `String`, for
+example `@__FILE__`.
+
+Multiple properties can be added to one testset, but if the same property is set on
+both parent and child testsets, the value in the parent testset takes precedence over
+that in the child.
+
+The suggested use of this macro is to place it inside a testset with unspecified type
+(see Examples). This will ensure that `Pkg.test` is unnaffected by the macro, but
+that the properties are added to the report when `TestReports.test` is used. This
+is because `TestReports` wraps package tests in a `ReportingTestSet`, and the macro
+only adds a property when it is within a `ReportingTestSet`. 
+
+If a child testset is a `ReportingTestSet` but its parent isn't, the property should
+be in the report when `TestReport.test` is used, assuming that the parent testset
+type doesn't do anything to affect the reporting behaviour. However this is not tested
+functionality.
+
+# Examples
+```julia
+using TestReports
+
+# Default testset used, so macro will not affect Pkg.test but will be used when
+# generating JUnit XML.
+@testset "MyTestSet" begin
+    @recordproperty "ID" "42"
+    @recordproperty "File" @__FILE__
+    @test 1==1
+    @test 2==2
+end
+```
+"""
+macro recordproperty(name::AbstractString, val::Union{AbstractString, Expr})
+    line = string(__source__.line)
+    file = string(__source__.file)
+    expr = quote
+        if typeof(get_testset()) == ReportingTestSet
+            if haskey(get_testset().properties, $name) 
+                @warn "Property " * $name * " already set and can't be set again in the same testset\nLine " * $line * ", " * $file
+            else
+                get_testset().properties[$name] = $val
```

Comment:
I wonder if we should do:
```suggestion
                get_testset().properties[$name] = repr($val)
```
or
```suggestion
                get_testset().properties[$name] = string($val)
```

to make things like numbers etc work

Reply:
Yeah `Any` is nice, didn't appreciate how much `EzXML` could handle

Reply:
I've updated the reference test example to include a few different value types.